django-stubs/db/models/fields/__init__.pyi

```diff
@@ -180,6 +181,7 @@ class Field(RegisterLookupMixin, Generic[_ST, _GT]):
         validators: Iterable[validators._ValidatorCallable] = ...,
         error_messages: _ErrorMessagesMapping | None = ...,
         db_comment: str | None = ...,
+        db_default: Any = ...,
```

Comment:
We can probably do better than `Any` for this type. According to documentation "This can be a literal value or a database function, such as Now." https://docs.djangoproject.com/en/5.0/ref/models/fields/#db-default

So something like `_ST | Expression | None`?

Reply:
I walked a lot around it, Any looks too abroad, for sure.
But I feel that I may not narrow user choices, bcause in Django's __init__ it looks next way.
https://github.com/django/django/blob/acfc7e3a735ffa41dcd9ad7f4f8fef97c1a2c3c6/django/db/models/fields/__init__.py#L221-L226
So technically it may be anything that follow `.resolve_expression` protocol or a thing, that may be wrapped in `Value`.
Django-stubs define `Value` as 
https://github.com/typeddjango/django-stubs/blob/29c74e6e5f2205a24d42aa2d6a0b596a0ccb9277/django-stubs/db/models/expressions.pyi#L186-L188
so by stub `Value` may be initialized with `value: Any`.

Thats why I some confused. Looks like db_default may be anything. Sure it may be narrowed to `_ST | Expression | None | NOT_PROVIDED`, but I have no idea is it correct or no. Please advice.

Reply:
But here as class attr it should be narrowed to:

```python 
class Field(RegisterLookupMixin, Generic[_ST, _GT]):
    ...
    db_default: NOT_PROVIDED | Expression
```
`__init__` passing expressions or wrap in Value, that is Expression subclass, wdyt?

Reply:
> anything that follow .resolve_expression protocol

Don't get too bogged down with "it could be anything", we need to apply a certain amount of pragmatism.

In practice `resolve_expression()` means `Expression` or `BaseExpression`. If users want to define their own DB functions/expressions, they should subclass `Expression`. This is consistent with many other definitions that accept Expression. If this were a problem in practice, we would probably have an issue about it, _then_ we could discuss alternatives (like using `Protocol` instead of concrete class).

```
BaseExpression
├── Expression
│   └── Func
│       ├── Now
│       └── ...
├── Query
└── Subquery
```

Since `Query` and `Subquery` probably aren't allowed or useful in the DEFAULT clause in any database, I'd go with `Expression` instead of `BaseExpression`.

> so by stub Value may be initialized with value: Any.

Yep, since `Value` doesn't know what type of column it will be used with.

But with the `Field` definition we know the type of field being defined (`CharField` should accept `str`, `IntegerField` should accept `int` etc). That's `_ST`.

Reply:
> But here as class attr it should be narrowed to:

Yes, you're right. Good point. `NOT_PROVIDED` -> `type[NOT_PROVIDED]` since it's using the class definition, not a class instance.